File: acme.cashflow.domain/DTO/Product/Compra.cs
using acme.cashflow.domain.DTO.Person;
using acme.cashflow.domain.DTO.Seguranca;
using acme.cashflow.domain.DTO.Util;
using System;
using System.Collections.Generic;
using System.Text;

namespace acme.cashflow.domain.DTO.Product
{
    public class Compra : AbstractEntity
    {
        public Compra()
        {
            ComprasProdutos = new HashSet<CompraProduto>();
        }

        public DateTime DataCompra { get; set; }
        public long ValorTotal { get; set; }
        public Guid UsuarioId { get; set; }
        public Guid FornecedorId { get; set; }
        public Guid CompetenciaId { get; set; }

        public virtual Usuario Usuario { get; set; }
        public virtual Fornecedor Fornecedor { get; set; }
        public virtual Competencia Competencia { get; set; }
        public virtual ICollection<CompraProduto> ComprasProdutos { get; set; }


    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/product/compra.ts
import { Fornecedor } from './../person/fornecedor';
import { UsuarioResponse } from './../security/usuario';
import { AbstractEntityCompetencia } from "../abstract-entity-competencia";
import { CompraProduto } from './compra-produto';

export class  Compra extends AbstractEntityCompetencia{
  public dataCompra : Date;
  public valorTotal: number;
  public usuarioId :string;
  public fornecedorId : string;

  public usuario : UsuarioResponse;
  public fornecedor: Fornecedor;
  public comprasProdutos: CompraProduto[];
}
